src/v/kafka/protocol/schemata/produce_request.json

```diff
@@ -35,10 +35,19 @@
   "fields": [
     { "name": "TransactionalId", "type": "string", "versions": "3+", "nullableVersions": "0+", "entityType": "transactionalId",
       "about": "The transactional ID, or null if the producer is not transactional." },
-    { "name": "Acks", "type": "int16", "versions": "0+",
-      "about": "The number of acknowledgments the producer requires the leader to have received before considering a request complete. Allowed values: 0 for no acknowledgments, 1 for only the leader and -1 for the full ISR." },
-    { "name": "TimeoutMs", "type": "int32", "versions": "0+",
-      "about": "The timeout to await a response in miliseconds." },
+    {
+      "name": "Acks",
+      "type": "int16",
+      "versions": "0+",
+      "about": "The number of acknowledgments the producer requires the leader to have received before considering a request complete. Allowed values: 0 for no acknowledgments, 1 for only the leader and -1 for the full ISR."
+    },
+    {
+      "name": "TimeoutMs",
+      "type": "int32",
+      "versions": "0+",
+      "default": "2147483647",
```

Comment:
would it be better to set this value to `-1` and handle it downstream it would be consistent with other fields like this .